tests/schema/tests.py

```diff
@@ -1709,3 +1709,39 @@ def test_add_textfield_unhashable_default(self):
         new_field.set_attributes_from_name("info")
         with connection.schema_editor() as editor:
             editor.add_field(Author, new_field)
+
+    @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific")
+    def test_alter_field_add_index_to_charfield(self):
+        # Create the table
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        # Ensure the table is there and has no index
+        self.assertNotIn(
+            "name",
+            self.get_indexes(Author._meta.db_table),
+        )
+        # Alter to add the index
+        old_field = Author._meta.get_field('name')
+        new_field = CharField(max_length=255, db_index=True)
+        new_field.set_attributes_from_name('name')
+        with connection.schema_editor() as editor:
+            editor.alter_field(Author, old_field, new_field, strict=True)
+        # Check that all the constraints are there
+        constraints = self.get_constraints(Author._meta.db_table)
+        name_indexes = []
+        for name, details in constraints.items():
+            if details['columns'] == ['name']:
+                name_indexes.append(details)
```

Comment:
I would make sure one of the index name ends with `_like` or simply iterate over `constraint.values()`.